__tests__/services/FunctionProcessor.test.ts

```diff
@@ -0,0 +1,83 @@
+import { processFunction } from "../../src/services/BlockService/FunctionProcessor";
+import { Block } from "../../src/types/Block";
+import { Params } from "../../src/types/Params";
+
+describe('processFunction', () => {
+    let params: Params;
+    let mockGoToPath: jest.Mock;
+    let mockSetTextAreaValue: jest.Mock;
+    let mockInjectMessage: jest.Mock;
+    let mockStreamMessage: jest.Mock;
+    let mockRemoveMessage: jest.Mock;
+    let mockEndStreamMessage: jest.Mock;
+    let mockShowToast: jest.Mock;
+    let mockDismissToast: jest.Mock;
+    let mockOpenChat: jest.Mock;
+
+    beforeEach(() => {
+        mockGoToPath = jest.fn();
+        mockSetTextAreaValue = jest.fn();
+        mockInjectMessage = jest.fn().mockResolvedValue(null);
+        mockStreamMessage = jest.fn().mockResolvedValue(null);
+        mockRemoveMessage = jest.fn().mockResolvedValue(null);
+        mockEndStreamMessage = jest.fn().mockResolvedValue(true);
+        mockShowToast = jest.fn();
+        mockDismissToast = jest.fn().mockReturnValue(null);
+        mockOpenChat = jest.fn();
+
+        params = {
+            userInput: "",
+            currPath: null,
+            prevPath: null,
+            goToPath: mockGoToPath,
+            setTextAreaValue: mockSetTextAreaValue,
+            injectMessage: mockInjectMessage,
+            streamMessage: mockStreamMessage,
+            removeMessage: mockRemoveMessage,
+            endStreamMessage: mockEndStreamMessage,
+            showToast: mockShowToast,
+            dismissToast: mockDismissToast,
+            openChat: mockOpenChat,
+        };
```

Comment:
You can inline `jest.fn()` directly inside params, there's no need to declare variables and do multiple assignments.